applications/services/gui/modules/file_browser.c

```diff
@@ -534,6 +566,7 @@ static bool file_browser_view_input_callback(InputEvent* event, void* context) {
                     if(event->key == InputKeyUp) {
                         model->item_idx =
                             ((model->item_idx - 1) + model->item_cnt) % model->item_cnt;
+                        model->animation_begin_frame = furi_get_tick();
```

Comment:
couple notes:

- View/ViewPort doesn't refresh itself. You need to send redraw request
- This code only works because of the other processes requesting redraw
- It make sense to add FuriTimer that will increment counter in the model

Reply:
Also there will be cases when file browser starts with long filename, looks like this code doesn't handles it.